cmd/maya-apiserver/app/server/volume_endpoint.go

```diff
@@ -19,15 +20,19 @@ const (
 
 // VolumeSpecificRequest is a http handler implementation. It deals with HTTP
 // requests w.r.t a single Volume.
-//
-// TODO
-//    Should it return specific types than interface{} ?
 func (s *HTTPServer) volumeSpecificRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-
-	fmt.Println("[DEBUG] Processing", req.Method, "request")
+	glog.Infof("received volume request: method '%s'", req.Method)
 
 	switch req.Method {
 	case "PUT", "POST":
+		// check the feature gate & switch if enabled
+		//
+		// NOTE:
+		//  feature gate is enabled/disabled for volume create request only
+		if util.CASTemplateFeatureGate() {
+			return s.volumeV1alpha1SpecificRequest(resp, req)
+		}
+
```

Comment:
If CAS gate is on then only we enter with CAS template way of volume creation. This is clear.
But I can see no provision for other handlers if CAS gate is on (Like GET for volume listing).
May be we do not want volume listing to go in CAS temlate's way (I dont know).
But then I can see that `volumeV1alpha1SpecificRequest` handler is again having other http verb for operation on volume resource like `GET` , `DELETE` .
This meant we never hit the `GET` and `DELETE` cases written in  `volumeV1alpha1SpecificRequest` handler as the only way to enter `volumeV1alpha1SpecificRequest` handler is `PUT` or `POST`

Reply:
Fine.